cmd/gaia/app/app.go

```diff
@@ -238,6 +240,34 @@ func (app *GaiaApp) initChainer(ctx sdk.Context, req abci.RequestInitChain) abci
 		panic(err) // TODO find a way to do this w/o panics
 	}
 
+	if len(genesisState.Txs) > 0 {
+		for _, genTx := range genesisState.Txs {
+			var tx auth.StdTx
+			err = app.cdc.UnmarshalJSON(genTx, &tx)
+			if err != nil {
+				panic(err)
+			}
+			bz, err := app.cdc.MarshalBinary(tx)
+			if err != nil {
+				panic(err)
+			}
+			res := app.BaseApp.DeliverTx(bz)
+			if !res.IsOK() {
```

Comment:
Hmm I don't know about this. Maybe we should allow the transactions to fail (and just not apply state changes, as `DeliverTx` already implements). Some people might submit invalid transactions - maybe they should just pay the fee.